### ISSUE TITLE
If a region file contains a huge number of entities, it is determined unreadable by mcworldlib

### ISSUE THREAD

**Issue by ievans3024** [MestreLion/mcworldlib#7]
What originally led me to this fantastic library was the fact that I had a world that kept freezing when certain areas loaded, and I needed a CLI/programming tool to try to inspect and fix it, because NBTExplorer was also freezing once I'd found the offending chunk and tried to drill into the chunk properties.

What I was able to see in NBTExplorer was that the offending chunk had 40233 entities listed. As it turns out, making a command block that sprays item entities (which I'd configured to have a lifespan of ~10 ticks) gets leaky in certain circumstances.

When I tried to load the region containing the offending chunk, this is the output I got:
```
Python 3.8.2 (default, Apr  8 2020, 14:31:25) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import mcworldlib as mc
>>> w = mc.load('new red castle broken')
>>> w.regions[-2,-1]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ian/PycharmProjects/mcworldlib/mcworldlib/region.py", line 67, in __getitem__
    self._regions[key] = RegionFile.load(region)
  File "/home/ian/PycharmProjects/mcworldlib/mcworldlib/region.py", line 114, in load
    return cls.parse(open(filename, 'rb'))
  File "/home/ian/PycharmProjects/mcworldlib/mcworldlib/region.py", line 150, in parse
    assert chunk.sector_count <= sector_count <= chunk.sector_count + 1, \
AssertionError: Length mismatch for region (-2, -1) in chunk (7, 15): region header declares 5 4096-byte sectors, but chunk data required 213.
```
Ultimately, just deleting and recreating the Entities list tag via NBTExplorer fixed the issue, and mcworldlib is now able to read the region, but I still wonder if this is indicative of a bug in this library.

**Comment by MestreLion** [comment]
Thanks for the compliments! "Fantastic"? I think my humble library is still too "young" and in prototype stage to deserve this, but still, thinks!

About the issue: this is not a bug _per se_, it seems your data is corrupt, as the `sector_count` for that chunk in the region "header" does not match the actual sectors count in the chunk data. And perhaps that's why both Minecraft and NBTExplorer crashed then you reading that chunk. I also remember chunks have a size limit (IIRC 1MB = 256 sectors), so maybe 213 was too close to that?

But in any case, my library could definitely handle this mismatch better than throwing an abrupt `AssertionError`. Maybe it could just spew out a *Warning* and move on, and use the chunk data size as the correct number of sectors. The mismatch would eventually be ironed out when writing the region file, as the region header index is rebuild from chunks actual data.

I'll leave this bug open until I fix this. Thanks for reporting!

**Comment by ievans3024** [comment]
That definitely sounds like a better way to handle that scenario. I'll work on a branch for that tonight.

**Comment by MestreLion** [comment]
Fixed in 9ef29f6429489d42120a075d8116abb7942628d6
